### ISSUE TITLE
Unrecognized directive [@this] on Larastan + Jetstream with Livewire installation

### ISSUE THREAD

**Issue by johnbacon** [Stillat/blade-parser#25]
I've created a reproduction repository: https://github.com/johnbacon/blade-parser-unrecognized-directive

The gist is this -- when `blade-parser` is used with PHPStan + Larastan and Jetstream with Livewire, a fatal error occurs:
```sh
johnbacon@Crunchy:~/www/blade-parser-unrecognized-directive$ php artisan blade:validate
Running 20 validators against 51 files; PHPStan document validation enabled.

   Stillat\BladeParser\Errors\Exceptions\CompilationException 

  Unrecognized directive [@this]

  at vendor/stillat/blade-parser/src/Compiler/Compiler.php:729
    725▕ 
    726▕                 if (method_exists($this, $compilationMethod)) {
    727▕                     $this->appendToBuffer($node, $this->{$compilationMethod}($node));
    728▕                 } else {
  ➜ 729▕                     throw new CompilationException("Unrecognized directive [@{$node->content}]");
    730▕                 }
    731▕             } elseif ($node instanceof PhpTagNode) {
    732▕                 $this->appendToBuffer($node, $node->content);
    733▕             } elseif ($node instanceof PhpBlockNode) {

      +17 vendor frames 

  18  artisan:37
```

The commands required to reproduce this issue are as follows:
- `composer create-project laravel/laravel blade-parser-unrecognized-directive`
- `composer require laravel/jetstream`
- `php artisan jetstream:install livewire`
- `composer require stillat/blade-parser`
- `composer require larastan/larastan:^2.0 --dev`

I have confirmed that adding `this` to `validation.php`'s ignore directives does not eliminate the issue:
```php
    'ignore_directives' => [
        'this',
    ],
```

Removing `larastan/larastan` does fix the issue. I have not tested it with PHPStan alone.

This error also leaves a number of `temp_*` files/folders in the root directory.

**Comment by JohnathonKoster** [comment]
Thanks for the report, and the repo! I'll take a look sometime this weekend to work on a fix

**Comment by JohnathonKoster** [comment]
This has been resolved with `1.6.1`. This release introduces a few changes to help improve the out of box experience related to this issue:

1. If a compilation failure happens, the temporary files will still be cleaned up
2. Directives inside `ignore_directives` will now disable parsing *and* compilation across all validators. Before this setting was only used on some specific validator implementations
3. Corrected a compilation error that would happen for some component tag input (wasn't causing the issue, but discovered after fixing it)
4. Custom Blade component compilers are now used by the validator, and should no longer have to add the `this` directive the ignore list
5. The `livewire` directive will be ignored by default when validating documents to prevent some issues due to lifecycle/service availability 

Happy coding!

**Comment by johnbacon** [comment]
Thanks John, both for the great fix and project!
